index.js

```diff
@@ -40,20 +45,28 @@ class KappaDrive {
   }
 
   _getDrive (metadata, content, cb) {
-    if (metadata === this.metadataKey) {
-      metadata = 'metadata'
-      content = 'content'
-    }
-    debug('getting drive feed', metadata, content)
-    this.core.feed(metadata, (err, metadata) => {
-      if (err) return cb(err)
-      this.core.feed(content, (err, content) => {
+    metadata = this._feeds[metadata] || this.core.feed(metadata)
+    content = this._feeds[content] || this.core.feed(content)
+
+    if (metadata && content) {
+      debug('got feeds', metadata, content)
+      var drive = hyperdrive(this._storage, { metadata, content })
+      drive.ready(() => cb(null, drive))
+    } else {
+      this.core.writer(METADATA, (err, metadata) => {
```

Comment:
`this.core` already contains this feed map. Does `core.feed` give what we need instead of having a separate map inside this module?

Reply:
Hah this comment should be at the top of this function xD

Reply:
See ac3e4e6

Reply:
So multifeed's `feed` fn takes a key, not a `localname`. We'd need to PR to get `core.feed` to return based on the localname or the key. Hence why I've used `core. _logs._feeds`. But it's not ideal.

Reply:
@noffle thoughts on this use case?

Reply:
Local names are purely local, and so they *must* be writers. So, you can use the `writer` API to fetch a feed by its localname.

Reply:
The challenge here is we don't know whether its a local feed or not, in `getDrive` we could be receiving a `localname` or a `pubKey` for another's feed. So in this case, having `core.feed(name)` would be really useful. Perhaps we should approach this differently here rather than changing multifeed... I'll have a think about it.

Reply:
There is some work ongoing closed source that attempts to address this issue. I am ok with merging this for now and then revisiting it later

Reply:
Ah! My bad -- I think I understand now. :)

What do you think about changing `_getDrive` to accept hypercore instances directly? Then:

- on line 76: use the `feed` api
- on line 108: use the `writer` api

and then send the resulting feed into `_getDrive`?

Reply:
My thoughts exactly yesterday :) but sadly ongoing hyperdrive changes mean this is in constant flux!